Content/Items/Breacher/Projectiles.BreachCannonSentry.cs

```diff
@@ -0,0 +1,585 @@
+        {
+			for (int projIndex = 0; projIndex < Main.projectile.Length; projIndex++) //super laser dealings
+			{
+				Projectile proj = Main.projectile[projIndex];
+
+				if (!proj.active || proj.type != Projectile.type || proj == Projectile)
+					continue;
+
+				var mp = proj.ModProjectile as BreachCannonSentry;
+
+				if (mp.superLaser || mp.superLaserContributer)
+					continue;
+
+				Vector2 mpTestEndpoint = mp.laserStartpoint + mp.CalculateOffset(mp.laserStartpoint, proj.rotation, 15, 1);
+				Vector2[] collisionPoint;
+
+				if (laserStartpoint == laserEndpoint || mp.laserStartpoint == mp.laserEndpoint)
+					collisionPoint = new Vector2[0];
+				else
+					collisionPoint = Collision.CheckLinevLine(laserStartpoint, laserEndpoint, mp.laserStartpoint, mpTestEndpoint);
+
+				if (collisionPoint.Length > 0)
+				{
+					mp.laserEndpoint = collisionPoint[0];
+					mp.superLaserContributer = true;
+
+					superLaser = true;
+					laserEndpoint = collisionPoint[0];
+					superCharge = 2;
+
+					Rectangle additionalCheck = new Rectangle((int)(laserEndpoint.X - 15), (int)(laserEndpoint.Y - 15), 30, 30);
+
+					float superAngle = MathHelper.Lerp(Projectile.rotation, Projectile.rotation + Helper.RotationDifference(proj.rotation, Projectile.rotation), 0.5f);
+					
+					for (int projIndex2 = 0; projIndex2 < Main.projectile.Length; projIndex2++)
+					{
+						Projectile proj2 = Main.projectile[projIndex2];
+						if (!proj2.active || proj2.type != Projectile.type)
+							continue;
+
+						var mp2 = proj2.ModProjectile as BreachCannonSentry;
+						if (mp2.superLaser || mp2.superLaserContributer)
+							continue;
+
+						float collisionPoint2 = 0f;
+
+						Vector2 mp2Offset = mp2.CalculateOffset(mp2.laserStartpoint, proj2.rotation, 15, 1);
+						if (mp2Offset == Vector2.Zero)
+							continue;
+
+						if (Collision.CheckAABBvLineCollision(additionalCheck.TopLeft(), additionalCheck.Size(), mp2.laserStartpoint, mp2.laserStartpoint + mp2Offset, 30, ref collisionPoint2))
+						{
+							mp2.superLaserContributer = true;
+							mp2.laserEndpoint = laserEndpoint;
+							superAngle = MathHelper.Lerp(superAngle, superAngle + Helper.RotationDifference(proj2.rotation, superAngle), 1.0f / superCharge);
+							superCharge++;
+						}
+					}
+
+
+					superLaserStartpoint = laserEndpoint;
+					superLaserEndpoint = superLaserStartpoint + CalculateOffset(superLaserStartpoint, superAngle, 15 * superCharge, superCharge);
+					break;
+				}
+			}
+		}
+
+		private void ManageCaches()
+		{
+			cache = new List<Vector2>();
+
+			for (int i = 0; i < 14; i++)
+			{
+				cache.Add(Vector2.Lerp(laserStartpoint, laserEndpoint, i / 14f));
+			}
+
+			cache.Add(laserEndpoint);
+
+			superCache = new List<Vector2>();
+
+			for (int i = 0; i < 14; i++)
+			{
+				superCache.Add(Vector2.Lerp(superLaserStartpoint, superLaserEndpoint, i / 14f));
+			}
+
+			superCache.Add(superLaserEndpoint);
+		}
+
+		private void ManageTrail()
```

Comment:
Seems a bit strange to have 4 trails for a straight line? Is there a possibility these could be either compressed into 1 shader?

Reply:
It's 2 trails per line. 2 are just for the superlaser, 2 are for the main laser. And while slightly inefficient, this makes it way easier to make changes to the lasers visuals on the fly. If absolutely nessecary, I could probably combine it so there's only two trails.